File: Wiki/C/code/C_Programming_A_Modern_Approach/ch06_Loops/projects/03.c
// Purpose: Reduce an input fraction to lowest terms.
//
//     Enter a fraction: <6/12>
//     In lowest terms: <1/2>

// Reference: page 123 (paper) / 148 (ebook)

#include <stdio.h>

    int
main(void)
{
    int numerator, denominator, m, n, r, gcd;

    printf("Enter a fraction: ");
    scanf("%d /%d", &numerator, &denominator);

    // let's compute the GCD of the fraction's numerator and denominator
    m = numerator;
    n = denominator;

    while (n != 0)
    {
        r = m % n;
        m = n;
        n = r;
    }

    gcd = m;

    // let's divide both the numerator and  denominator by the GCD to reduce the
    // fraction to its lowest terms
    printf("In lowest terms: %d/%d\n", numerator / gcd, denominator / gcd);

    return 0;
}

File: Wiki/lua/code/Programming_in_Lua_3rd_edition/ch02_Types_and_Values/exercises/03.lua
-- Purpose: The  number `12.7`  is equal  to  the fraction  `127/10`, where  the
-- denominator is a power of ten.  Can you express it as a common fraction where
-- the denominator is a power of two?  What about the number `5.5`?

-- Reference: page 18 (paper) / 37 (ebook)


-- `12.7` cannot  be expressed as a  common fraction where the  denominator is a
-- power of two, because:
--
--    - the prime factor decomposition of `10` is `2 * 5`
--    - `127` is a prime number
--
-- For the denominator to be a power of two, we would need to get rid of `5`.
-- That is, we would need to divide the numerator `127` by `5`.
-- We  can't, because  `127` is  prime  (i.e. can  only  be divided  by `1`  and
-- itself).
--
-- ---
--
-- `5.5` is equal  to the fraction `11/2`.   The denominator in the  latter is a
-- power of two.
